arkane/ess/terachem.py

```diff
@@ -57,8 +57,31 @@ class TeraChemLog(ESSAdapter):
     TeraChemLog is an adapter for the abstract class ESSAdapter.
     """
 
-    def __init__(self, path):
+    def __init__(self, path, check_for_errors=True):
         self.path = path
+        if check_for_errors:
+            self.check_for_errors()
+
+    def check_for_errors(self):
+        """
+        Checks for common errors in a TeraChem log file.
+        If any are found, this method will raise an error and crash.
+        """
```

Comment:
If possible, can you add a unit test for TeraChem?

Reply:
I haven't used TeraChem before so I don't have any failed jobs, and it seems like ARC doesn't have any either. Feel free to send me some to add if you have them though